digicampipe/io/event_stream.py

```diff
@@ -1,25 +1,30 @@
 from digicampipe.io import zfits, hdf5
 
 
-def event_stream(file_list, camera_geometry, camera, expert_mode=False, max_events=None, mc=False):
-
-    for file in file_list:
-
-        if not mc:
-
-            data_stream = zfits.zfits_event_source(url=file,
-                                                   expert_mode=expert_mode,
-                                                   camera_geometry=camera_geometry,
-                                                   max_events=max_events,
-                                                   camera=camera)
-        else:
-
-            data_stream = hdf5.digicamtoy_event_source(url=file,
-                                                       camera_geometry=camera_geometry,
-                                                       camera=camera,
-                                                       max_events=max_events)
-
+def event_stream(
+    filelist_or_url,
+    camera=None,
+    camera_geometry=None,
+    expert_mode=None,
+    max_events=None,
+    mc=False
+):
+    if isinstance(filelist_or_url, (str, bytes)):
```

Comment:
FYI there is something similar for numpy array and single values : 

https://docs.scipy.org/doc/numpy/reference/generated/numpy.atleast_1d.html

I was quickly looking if there is something similar for strings and lists of strings but could not find it...

Reply:
Yes, this is an ugly point, it makes the interface "unclear". 
A function should have a clear interface. 

But, I thought this function is very exposed to the "end-user" so adding this feature felt "okay-ish" ... 

I'm not sure anymore though... maybe its fine

Reply:
Yes maybe a bit "hard" to read but I suppose a comment would make it easier

Reply:
also you can overwrite the variables right? 

just removing the else statement and: 

```python
def event_stream(
    filelist,
   ....
   )
...
   if isinstance(filelist, (str, bytes)):
      filelist = [filelist]

...
```

Reply:
:-D ... I guess you know my opinion about comments. When they are needed, the interface is broken :-D